### PR TITLE
rustdesk-flutter: unstable-2024-02-03 -> 1.2.3-unstable-2024-02-11

### PR DESCRIPTION
**Pull request by dasJ**
Also multiple fixups:
- Fix strucuted attrs
- Switch to Flutter 3.16
- Add some comments on how to generate files
- Make patches apply more neatly
- Use proper cc
- Make output hash independant of metadata
- Softcode .so file ending

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ru/rustdesk-flutter/package.nix [hunk 1]
```diff
@@ -134,8 +139,12 @@ in flutter313.buildFlutterApplication rec {
     xdotool
   ];
 
-  postPatch = ''
+  prePatch = ''
     chmod -R +w ..
+  '';
+  patchFlags = "-p1 -d ..";
```
Comment (SuperSandro2000) [inline pkgs/by-name/ru/rustdesk-flutter/package.nix:145]:
```suggestion
  patchFlags = [ "-p1" "-d" ".." ];
```
that should be an array

File: pkgs/by-name/ru/rustdesk-flutter/package.nix [hunk 2]
```diff
@@ -36,22 +37,24 @@
...
   };
 
   strictDeps = true;
-  strucutedAttrs = true;
+  __strucutedAttrs = true;
 
   # Configure the Flutter/Dart build
   sourceRoot = "source/flutter";
-  pubspecLock = lib.importJSON ./pubspec.lock.json;
+  pubspecLock = lib.importJSON ./pubspec.lock.json; # curl https://raw.githubusercontent.com/rustdesk/rustdesk/16db977fd81e14af62ec5ac7760a7661a5c24be8/flutter/pubspec.lock | yq
```
Comment (SuperSandro2000) [inline pkgs/by-name/ru/rustdesk-flutter/package.nix:57]: nit: maybe write it above this line to not make it so long

**Review by SuperSandro2000 — APPROVED** [review]
LGTM

**Review by SuperSandro2000 — APPROVED** [review]
diff LGTM, unfortunately linking fails on aarch64-linux and I can make much out of that error :/

**Review by 0x4A6F — APPROVED** [review]

**Comment by Thatoo** [comment]
could it be backported to 23.11?

**Comment by dasJ** [comment]
Unlikely, the package isn't even present on 23.11 afaik

**Comment by dasJ** [comment]
No, rustdesk-flutter

**Comment by Thatoo** [comment]
Well that is exactly the reason I ask if it could be backported to make it exists on 23.11.

**Comment by dasJ** [comment]
Ah, one could do that by backporting both #286129 and this one.

**Comment by Thatoo** [comment]
Ah, ok, I see, well then I'll ask first to the other one.

**Pull request merged**